test/f3_test.go

```diff
@@ -31,134 +32,89 @@ const (
 var log = logging.Logger("f3-testing")
 
 func TestSimpleF3(t *testing.T) {
-	ctx := context.Background()
 	env := newTestEnvironment(t, 2, false)
 
-	env.Connect(ctx)
-	env.run(ctx)
-	env.waitForInstanceNumber(ctx, 5, 10*time.Second, false)
```

Comment:
The context is now internalized in `env` (and it gets canceled when the test ends).